ast/desugar/Desugar.cc

```diff
@@ -1128,6 +1141,15 @@ TreePtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what) {
                 TreePtr nil = MK::Send1(zeroLengthRecvLoc, ast::MK::Constant(zeroLengthLoc, core::Symbols::Magic()),
                                         core::Names::nilForSafeNavigation(), MK::Local(zeroLengthRecvLoc, tempRecv));
                 auto iff = MK::If(zeroLengthLoc, std::move(cond), std::move(nil), std::move(send));
+
+                if (chainLvarName.exists()) {
+                    cond = MK::Send1(zeroLengthLoc, ast::MK::Constant(zeroLengthRecvLoc, core::Symbols::NilClass()),
```

Comment:
> It would be great to find a way to structure the code such that Sorbet doesn't need to keep re-checking `NilClass === x` every time.

I agree; I haven't figured out a good way to do that though 😞

Reply:
@SeekingMeaning Can I propose you an idea?

What if you implement this:

https://github.com/sorbet/sorbet/issues/3397

and then use that to implement this desugaring?